robot/control/main.cpp

```diff
@@ -145,51 +195,12 @@ int main() {
     moduleList.emplace_back(curTime, KickerModule::period,        KickerModule::runtime,        &kicker);
     moduleList.emplace_back(curTime, BatteryModule::period,       BatteryModule::runtime,       &battery);
     moduleList.emplace_back(curTime, RotaryDialModule::period,    RotaryDialModule::runtime,    &dial);
-    moduleList.emplace_back(curTime, LEDModule::period,           LEDModule::runtime,           &led);
+    moduleList.emplace_back(curTime, LEDModule::period,           LEDModule::runtime,           led.get());
 
-    while (true) {
-        // Must do all timing in systick
-        // Base time we use is in systick, and if you convert from systick
-        // to us and try to compare timepoints, the value will "overflow" at
-        // some random point which messes up the `Current - Start` magic
-        // that automatically deals with delta time using the math behind overflows
-        uint32_t loopStartTime = DWT_GetTick();
-        uint32_t loopEndTime = DWT_GetTick() + SUPER_LOOP_PERIOD*DWT_SysTick_To_us();
+    osThreadDef(SuperLoopTask, superLoopTask, osPriorityAboveNormal, 1, 1 << 15);
```

Comment:
Do you have any docs for this call? I'm only seeing the 4 arg version in the keil cmsis docs here. https://www.keil.com/pack/doc/CMSIS/RTOS/html/group__CMSIS__RTOS__ThreadMgmt.html#gaee93d929beb350f16e5cc7fa602e229f

Reply:
Nope. I looked at the header file - the version that we have is:

```cpp
#define osThreadDef(name, thread, priority, instances, stacksz)
```

Reply:
As a side note, I mentioned on slack that it might be a good idea to use FreeRTOS APIs instead of the CMSIS wrapper layer - they are certainly better documented, which might be a point in favor.